flumeserver/indexer/indexer.go

```diff
@@ -125,263 +142,92 @@ func applyParameters(query string, params ...interface{}) string {
-        signer = types.NewLondonSigner(tx.ChainId())
-        accessListRLP, _ = rlp.EncodeToBytes(tx.AccessList())
-        gasPrice = tx.GasFeeCap().Uint64()
-      default:
-        signer = types.NewEIP155Signer(tx.ChainId())
-      }
-      sender, _ := types.Sender(signer, tx)
-      var to []byte
-      if tx.To() != nil {
-        to = trimPrefix(tx.To().Bytes())
-      }
-      v, r, s := tx.RawSignatureValues()
-      statements := []string{}
-      // If this is a replacement transaction, delete any it might be replacing
-      statements = append(statements, applyParameters(
-        "DELETE FROM mempool.transactions WHERE sender = %v AND nonce = %v",
-        sender,
-        tx.Nonce(),
-      ))
-      // Insert the transaction
-      statements = append(statements, applyParameters(
-        "INSERT INTO mempool.transactions(gas, gasPrice, hash, input, nonce, recipient, `value`, v, r, s, sender, `type`, access_list, gasFeeCap, gasTipCap) VALUES (%v, %v, %v, %v, %v, %v, %v, %v, %v, %v, %v, %v, %v, %v, %v)",
-        tx.Gas(),
-        gasPrice,
-        txHash,
-        getCopy(compress(tx.Data())),
-        tx.Nonce(),
-        to,
-        trimPrefix(tx.Value().Bytes()),
-        v.Int64(),
-        r,
-        s,
-        sender,
-        tx.Type(),
-        compress(accessListRLP),
-        trimPrefix(tx.GasFeeCap().Bytes()),
-        trimPrefix(tx.GasTipCap().Bytes()),
-      ))
-      // Delete the transaction we just inserted if the confirmed transactions
-      // pool has a conflicting entry
-      statements = append(statements, applyParameters(
-        "DELETE FROM mempool.transactions WHERE sender = %v AND nonce = %v AND (sender, nonce) IN (SELECT sender, nonce FROM transactions)",
-        sender,
-        tx.Nonce(),
-      ))
-      if txCount > (11 * mempoolSlots / 10) {
-        // More than 10% above mempool limit, prune some.
-        statements = append(statements, applyParameters(
-          "DELETE FROM mempool.transactions WHERE gasPrice < (SELECT gasPrice FROM mempool.transactions ORDER BY gasPrice LIMIT 1 OFFSET %v)",
-          mempoolSlots,
-        ))
-        txCount = mempoolSlots
-      }
-      if _, err := db.Exec(strings.Join(statements, " ; ") + ";"); err != nil {
-        log.Printf("Error on insert: %v - '%v'", err.Error(), strings.Join(statements, " ; ") )
-      }
-      txCount++
-    case chainEvent := <- ch:
-      start := time.Now()
-      BLOCKLOOP:
+	db.Exec("DELETE FROM mempool.transactions WHERE 1;")
+	for {
+		select {
+	case <-quit:
+		if !processed {
+			panic("Shutting down without processing any blocks")
+			//try not to panic, pass back into main and do proper cleanup
+		}
+		log.Printf("Shutting down index process")
+		return
+	case <- pruneTicker.C:
+		mempool_indexer1(db, mempoolSlots, processed, txCount, txDedup)
+	case tx := <-txCh:
+		mempool_indexer2(db, mempoolSlots, processed, txCount, txDedup, tx)
+	case chainEvent := <- ch:
```

Comment:
It looks like you're still supporting the legacy indexer. Right now that's fine, but I want to make sure its removal is on your list.